server/http_test.go

```diff
@@ -293,3 +295,76 @@ func TestParseRequestURL(t *testing.T) {
 		}
 	}
 }
+
+type fakeCache struct {
```

Comment:
I think it would be nice to add a generic in-memory cache implementation that can be initialized with a map of contents. I think this can be done in a follow up step though.